.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,69 @@
+name: Barretenberg Benchmarks
+
+# Only run on push to master
+on:
+  push:
+    branches:
+      - master
+
+# This will cancel previous runs when a branch or PR is updated
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  bberg-bench:
+    name: Barretenberg Benchmarks
+    runs-on: ubuntu-latest # run in linux environment
+
+    steps:
+
+      - name: Checkout barretenberg
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }} # checkout HEAD of triggering branch
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+     # Only run in Linux environment for now
+      - name: Setup Linux environment
+        run: |
+          sudo apt update
+          sudo apt install libomp-dev cmake ninja-build
+
+      - name: Download SRS elements
+        working-directory: cpp/srs_db
+        run: ./download_ignition.sh 3 # only download first 4 transcript files
+        
+      - name: Build Honk benchmarks
+        working-directory: cpp
+        run: |
+          cmake --preset bench
+          cmake --build --preset bench --target ultra_honk_bench
+      
+      - name: Run Honk benchmarks
+        working-directory: cpp/build-bench
+        run: |
+          bin/ultra_honk_bench --benchmark_format=json > ../src/barretenberg/benchmark/honk_bench/bench_results.json
+      
+      # Utilize github-action-benchmark to automatically update the plots at 
+      # https://aztecprotocol.github.io/barretenberg/dev/bench/ with new benchmark data.
+      # This also creates an alert if benchmarks exceed the threshold specified below.
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: C++ Benchmark
+          tool: 'googlecpp'
+          output-file-path: cpp/src/barretenberg/benchmark/honk_bench/bench_results.json
+          # Access token to deploy GitHub Pages branch
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Push and deploy GitHub pages branch automatically
+          auto-push: true
+          # Enable Job Summary for PRs
+          summary-always: true
+          # Show alert with commit comment on detecting possible performance regression
+          alert-threshold: '120%' # alert if bench result is 1.2x worse
```

Comment:
Will this be enough if there are incremental changes that drain performance?

Reply:
To be honest I'm viewing the GH Action portion of this work as just a nice to have with pretty pictures. I think it needs to be incumbent upon the developer to manually check the benchmarks in their branch any time they suspect a change. I suppose we could also automate running the benchmarks and comparing with master but the dev would still have to manually check the result

Reply:
It's a nice facility, so why not use it? If it's possible to add several alerts, I'd do a 90% alert, too. It can be even more disturbing if the circuit shrinks for no reason

Reply:
Looking at the docs it seems that there's no way to have an upper and lower bound for the alert. The nice thing about having the autogenerated plots is that unexpected changes will not go unnoticed and the offending commit is displayed. Maybe I'll change this threshold to 1.05x and we'll see how that goes. I'm not sure what the random variation will be